subsys/net/ip/net_context.c

```diff
@@ -961,40 +995,16 @@ static int sendto(struct net_pkt *pkt,
 	}
 #endif /* CONFIG_NET_OFFLOAD */
 
-	switch (net_context_get_ip_proto(context)) {
-	case IPPROTO_UDP:
-#if defined(CONFIG_NET_UDP)
-		/* Bind default address and port only if UDP */
-		ret = bind_default(context);
-		if (ret) {
-			return ret;
-		}
-
-		ret = create_udp_packet(context, pkt, dst_addr, &pkt);
-#endif /* CONFIG_NET_UDP */
-		break;
-
-	case IPPROTO_TCP:
-		ret = net_tcp_queue_data(context, pkt);
-		break;
-
-	default:
-		ret = -EPROTONOSUPPORT;
-	}
-
-	if (ret < 0) {
-		if (ret == -EPROTONOSUPPORT) {
-			NET_DBG("Unknown protocol while sending packet: %d",
-				net_context_get_ip_proto(context));
-		} else {
-			NET_DBG("Could not create network packet to send (%d)",
-				ret);
-		}
-
-		return ret;
+#if defined(CONFIG_NET_TLS) || defined(CONFIG_NET_DTLS)
+	if (context->options.tls) {
```

Comment:
Do you mean that for "TLS sockets", you're going to randomly ignore "cb" and "user_data" parameters?

(Layering violation. There's no TLS on "net_context" level. And can be only by making bunch of slips like above, compromising the original net_context API contract (which is not ideal in the first place, but at least "stable")).

Reply:
Oh, "timeout" is being ignored too, re: https://github.com/zephyrproject-rtos/zephyr/pull/7118/commits/f8f1f33048a21b7e921f88d0444c14d6000af87a#r184029293